src/app.tsx

```diff
@@ -0,0 +1,438 @@
+    backgroundColor: "rgb(62, 122, 235)",
+    height: "10px",
+    borderRadius: "1rem",
+    width: `${progressBarLength}px`,
+    // animation
+    transition: "1s ease",
+  };
+
+  return (
+    <div className="absolute top-16 left-16 z-10">
+      <div style={progressBarStyle}>
+        <div style={filledProgressBarStyle} />
+      </div>
+    </div>
+  );
+};
+
+interface GameProps {}
+
+// main parent component
+const Game: React.FC<GameProps> = (props) => {
+  // define states using state hook
+  // typescript implicitly knows the type of each state from initial value
+  const [showModal, setShowModal] = useState(false);
+  const [currentBugType, setCurrentBugType] = useState(""); // so that modal component knows a bug is good or bad
+  const [hasWon, setHasWon] = useState(false);
+  const [correctBugs, setCorrectBugs] = useState(new Set()); // tracking progress
+  const [bugIds, setBugIds] = useState<number[]>([]); // an array of bug ids
+
+  const intervalID = useRef(0);
+
+  // this variable keeps track of total # of spawned bugs
+  // also sets the id for a newest spawned bug
+  const numSpawnedBugs = useRef(0);
+
+  // component did mount
+  useEffect(() => {
+    // create a new bug and give it a unique index
+
+    const spawnNewBug = () => {
+      // if (n > UPPER_COUNT) return
+      if (bugIds.length >= UPPER_BUG_COUNT) {
+        return;
+      } else {
+        numSpawnedBugs.current += 1;
+        setBugIds((prevState) => {
+          return [...prevState, numSpawnedBugs.current];
+        });
+      }
+    };
+
+    intervalID.current = window.setInterval(spawnNewBug, 1000);
+
+    return () => {
+      clearInterval(intervalID.current);
+    };
+  }, [bugIds, numSpawnedBugs]);
+
+  /**
+   * toggles the render state of modal component
+   */
+  const toggleModal = () => {
+    setShowModal((prevState) => !prevState);
+  };
+
+  /**
+   * Delivers info about current clicked bug to Modal and removes the bug from screen
+   *
+   * @param bugId - the id of the bug so that it can be deleted from bugs array
+   */
+  const handleBug = (bugType: string, bugId: number) => {
+    toggleModal();
+
+    setCurrentBugType(bugType);
+
+    if (BUGS[bugType]) {
+      if (!BUGS[bugType].isFriendly) {
+        if (!hasWon) {
+          if (!correctBugs.has(bugType)) {
+            // does not have bugId yet
+            if (correctBugs.add(bugType).size === TARGET_CORRECT_GUESSES) {
+              // check for win on next guess
+              // if win, don't reset yet so that progress bar shows full right after winning
+              setHasWon(true);
+              setCorrectBugs((prevState) => {
+                return new Set([...prevState, bugType]);
+              });
+            } else {
+              // if next guess not win, just add to list
+              setHasWon(false);
+              setCorrectBugs((prevState) => {
+                return new Set([...prevState, bugType]);
+              });
+            }
```

Comment:
Move this to a separate function `handleCorrectBugClick`

```
if (!hasWon) {
          if (!correctBugs.has(bugType)) {
            // does not have bugId yet
            if (correctBugs.add(bugType).size === TARGET_CORRECT_GUESSES) {
              // check for win on next guess
              // if win, don't reset yet so that progress bar shows full right after winning
              setHasWon(true);
              setCorrectBugs((prevState) => {
                return new Set([...prevState, bugType]);
              });
            } else {
              // if next guess not win, just add to list
              setHasWon(false);
              setCorrectBugs((prevState) => {
                return new Set([...prevState, bugType]);
              });
            }
```